File: bench/dotnet/http.cs
namespace Bench;
static class Http
{
    // Concurrent HTTP: fire N GETs at the local server (each sleeps ~20ms
    // server-side) and count the 200s — async/await on the thread pool, all N
    // in flight at once. MaxConnectionsPerServer = N so nothing queues (the
    // analog of node's agent:false). Checksum = N.
    public static async Task Run(int n)
    {
        var handler = new SocketsHttpHandler { MaxConnectionsPerServer = n };
        using var client = new HttpClient(handler);
        var tasks = new Task<int>[n];
        for (int i = 0; i < n; i++)
        {
            tasks[i] = Task.Run(async () =>
            {
                try
                {
                    using var resp = await client.GetAsync("http://127.0.0.1:8089/");
                    return resp.IsSuccessStatusCode ? 1 : 0;
                }
                catch { return 0; }
            });
        }
        var results = await Task.WhenAll(tasks);
        Console.WriteLine(results.Sum());
    }
}

File: bench/clojure/http.clj
;; N concurrent GETs to a local server; count the 200s. java.net.http + futures.
(import '[java.net.http HttpClient HttpRequest HttpResponse$BodyHandlers]
        '[java.net URI])
(def n (Long/parseLong (or (System/getenv "BENCH_N") "100")))
(def port (or (System/getenv "BENCH_HTTP_PORT") "8089"))
(def url (str "http://127.0.0.1:" port "/"))
(def client (HttpClient/newHttpClient))
(defn fetch [_]
  (try
    (let [req (.build (.uri (HttpRequest/newBuilder) (URI/create url)))
          resp (.send client req (HttpResponse$BodyHandlers/ofString))]
      (if (= 200 (.statusCode resp)) 1 0))
    (catch Exception _ 0)))
(println (reduce + (map deref (doall (map (fn [i] (future (fetch i))) (range n))))))

;; Stop the agent/future thread pool so the JVM exits promptly (else it lingers ~60s).
(shutdown-agents)
